projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-transient.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * REST API endpoint for editing Jetpack Transients.
+ *
+ * @package automattic/jetpack
+ * @since 9.7.0
+ */
+
+/**
+ * Edit and delete transients.
+ *
+ * @since 9.7.0
+ */
+class WPCOM_REST_API_V2_Endpoint_Transient extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'transients';
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 *
+	 * /sites/$site/transients/$name/delete
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<name>\w{1,172})/delete/',
+			array(
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'delete_transient' ),
+					'permission_callback' => 'is_user_logged_in',
```

Comment:
we might improve the permission callback here. For instance, asking for a Simple site (I'm not sure we should ban them btw). Anyway, you can pass an array to call the class-method handler:

```suggestion
					'permission_callback' => array( $this, 'permission_callback' ),
```

and then add the new method:

```php
public function permission_callback( \WP_REST_Request $request ) {		
	return is_user_logged_in();
}
```

Reply:
TBD: do we need any additional permissions here?

Reply:
Could we get your opinion here, @obenland?

we didn't find any capability specific to transient, but might use another one that could work, for instance `manage_options`.

Reply:
`read` is the most basic capability that all roles should have: https://wordpress.org/support/article/roles-and-capabilities/#read

> Where possible, you should always use `current_user_can`; instead of checking if the user is logged in (authentication), check whether they can perform the action (authorization).
—https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-custom-endpoints/#permissions-callback

Reply:
Thanks! I updated the permissions check to `return current_user_can( 'read' );`: https://github.com/Automattic/jetpack/pull/19632/commits/f25b62d0e0888cb056f83151c4d74a261a98efed